File: CSE222 - Data Structures and Algorithms/hw/hw04/src/AnsiEscape.java
package src;

public class AnsiEscape {
    public static enum Color {
        BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE, DEFAULT
    }

    /** Deletes all the data at the screen */
    public static void clearScreen() {
        System.out.printf("\u001B[2J");
    }

    /**
     * Sets the cursor at given (row, col) location
     * (O, 0) is the top left, (row, col) is the right bottom
     * @param row
     * @param col
     */
    public static void setCursor(int row, int col) {
        System.out.printf("\033[%d;%dH", row, col);
    }

    /**
     * Sets font color of the screen 
     * @param c font color
     */
    public static void setFGColor(AnsiEscape.Color c) {
        String colorCode = null;
        switch (c) {
            case BLACK:
                colorCode = "\u001B[30m";
                break;
            case RED:
                colorCode = "\u001B[31m";
                break;
            case GREEN:
                colorCode = "\u001B[32m";
                break;
            case YELLOW:
                colorCode = "\u001B[33m";
                break;
            case BLUE:
                colorCode = "\u001B[34m";
                break;
            case MAGENTA:
                colorCode = "\u001B[35m";
                break;
            case CYAN:
                colorCode = "\u001B[36m";
                break;
            case WHITE:
                colorCode = "\u001B[37m";
                break;
            case DEFAULT:
                colorCode = "\u001B[0m";
                break;
        }
        System.out.printf("%s", colorCode);
    }

    /**
     * Sets BG color of the screen
     * @param c background color
     */
    public static void setBGColor(AnsiEscape.Color c) {
        String colorCode = null;
        switch (c) {
            case BLACK:
	            colorCode ="\u001B[40m";
                break;
            case RED:
	            colorCode = "\u001B[41m";
                break;
            case GREEN:
	            colorCode ="\u001B[42m";
                break;
            case YELLOW:
	            colorCode ="\u001B[43m";
                break;
            case BLUE:
	            colorCode = "\u001B[44m";
                break;
            case MAGENTA:
	            colorCode ="\u001B[45m";
                break;
            case CYAN:
	            colorCode = "\u001B[46m";
                break;
            case WHITE:
	            colorCode ="\u001B[47m";
                break;
            case DEFAULT:
                colorCode = "\u001B[0m";
                break;
        }
        System.out.printf("%s", colorCode);
    }
}

File: cse241-object-oriented-programming/hw/hw05/src/ansiEscape.cpp
#include <iostream>
#include "ansiEscape.h"

using std::cout;
using std::endl;

namespace AnsiEscape {
    void clearScreen () {
        cout << "\x1B[2J";
    }

    void setCursor (int row, int col) {
        // printf("\033[%d;%dH", Y, X)
        cout << "\033[" << row << ";" << col << "H";
    }

    void setFontColor (Color c) {
        switch (c) {
            case Color::red:
                cout << "\x1B[1;31m"; break;
            case Color::green:
                cout << "\x1B[1;32m"; break;
            case Color::yellow:
                cout << "\x1B[1;33m"; break;
            case Color::blue:
                cout << "\x1B[1;34m"; break;
            case Color::magenta:
                cout << "\x1B[1;35m"; break;
            case Color::cyan:
                cout << "\x1B[1;36m"; break;
            case Color::white:
                cout << "\x1B[1;37m"; break;
            case Color::normal:
            default:
                cout << "\x1B[0m";
        }
    }

    void setBackgroundColor (Color c) {
        switch (c) {
            case Color::red:
                cout << "\x1B[41m"; break;
            case Color::green:
                cout << "\x1B[42m"; break;
            case Color::yellow:
                cout << "\x1B[43m"; break;
            case Color::blue:
                cout << "\x1B[44m"; break;
            case Color::magenta:
                cout << "\x1B[45m"; break;
            case Color::cyan:
                cout << "\x1B[46m"; break;
            case Color::white:
                cout << "\x1B[47m"; break;
            case Color::normal:
            default:
                cout << "\x1B[0m";
        }
    }
}